backend/src/API/teamAPI.ts

```diff
@@ -106,6 +131,12 @@ export const setTeam = async (teamBody: Team, member: IdolMember): Promise<Team>
   return teamBody;
 };
 
+/**
+ * Deletes a team by removing all members, leaders, and formerMembers.
```

Comment:
Tracing this down, it seems like additionally, it removes all references between the member, leaders, and formerMembers from this team. I.e. member, leader, formerMember will no longer have his team under `subteams` field and `formerSubteams` field.

Maybe add that in the docstring?